ProcessLib/BalanceProcess/BalanceProcess.cpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "BalanceProcess.h"
+
+#include <cassert>
+
+#include "ProcessLib/Utils/CreateLocalAssemblers.h"
+
+namespace ProcessLib
+{
+BalanceProcess::BalanceProcess(MeshLib::Mesh& boundary_mesh,
+                               std::size_t bulk_property_number_of_components)
+{
+    DBUG("Create local balance assemblers.");
+    // Populate the vector of local assemblers.
+    _local_assemblers.resize(boundary_mesh.getElements().size());
+
+    // needed to create dof table
+    std::unique_ptr<MeshLib::MeshSubset const> mesh_subset_all_nodes(
+        new MeshLib::MeshSubset(boundary_mesh, &boundary_mesh.getNodes()));
+
+    // Collect the mesh subsets in a vector.
+    std::vector<std::unique_ptr<MeshLib::MeshSubsets>> all_mesh_subsets;
+    std::generate_n(
+        std::back_inserter(all_mesh_subsets),
+        bulk_property_number_of_components,
+        [&]() {
+            return std::unique_ptr<MeshLib::MeshSubsets>{
+                new MeshLib::MeshSubsets{mesh_subset_all_nodes.get()}};
+        });
+
+    // needed for creation of local assemblers
+    std::unique_ptr<NumLib::LocalToGlobalIndexMap const> dof_table(
+        new NumLib::LocalToGlobalIndexMap(std::move(all_mesh_subsets),
+                                          NumLib::ComponentOrder::BY_LOCATION));
+
+    boost::optional<MeshLib::PropertyVector<std::size_t> const&>
+        bulk_element_ids(boundary_mesh.getProperties()
+                             .template getPropertyVector<std::size_t>(
+                                 "OriginalSubsurfaceElementIDs"));
```

Comment:
If I change the name here, I have to change also the name in the tool ExtractSurface. If you really want change the name I would do this in a separate PR.